core/database/worker.py

```
@@ -43,17 +53,41 @@ async def worker(self):
+
+                    # Only commit once, this way it doesn't take up much cpu while avoiding unnecessary restrictions.
                     await db.commit()
-                    # print("success!")
+
+                    for f in succeeded:
+                        """
+                        `f` here is the future. When await is required, Future is passed alongside the queue. Once done
+                        """
+                        if f is not None:
+                            f.set_result(True)
                 except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)